build.gradle.kts

```diff
@@ -78,24 +83,49 @@ kotlin {
             }
         }
 
-        named("jvmMain") {
+        val jvmMain by getting {
+
             dependencies {
                 implementation("org.snakeyaml:snakeyaml-engine:2.6")
             }
         }
 
-        named("jvmTest") {
+        val jvmTest by getting {
+            dependsOn(commonTest)
+
             dependencies {
                 implementation("io.kotest:kotest-runner-junit5:5.6.2")
             }
         }
 
-        named("jsMain") {
+        val nonJVMMain by creating {
+            dependsOn(commonMain)
```

Comment:
Should this be added to the `jvmMain` set as well?

Reply:
No, to my understanding, all default source sets (the ones retrieved by `by getting`) implicitly depend on commonMain. We have to do it for nonJVMMain because we are creating it as a new source set. We then have to specify that all the non JVM source sets depend on this one. See @aSemy's [review](https://github.com/charleskorn/kaml/pull/458#discussion_r1308459399) that highlighted this to me.

There's a good example of this [here](https://kotlinlang.org/docs/multiplatform-compatibility-guide.html#deprecated-api-for-adding-kotlin-source-sets-directly-to-the-kotlin-compilation) under *What's the best practice now?*

Reply:
Makes sense, thanks for clarifying.